### PR TITLE
Documentation Schema

### LINKED ISSUE #37: Add minimal documentation
Provide the documentation skeleton

### LINKED ISSUE #65: Write definition of done, PR template
We could improve the PR quality by providing a simple template with checks to do.
Checks may include:
- documentation
- CI
- tests / lint
- code quality

### PR DESCRIPTION
**Pull request by hugoShaka**
This includes all the files to be able to write and compile documentation.
This closes #37 and closes #65

### CODE REVIEW

**Comment by hugoShaka** [comment]
Fixes #37

File: docs/mail.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
+Email basics
+============
+
+The big picture
+---------------
+Sending email involves lots of components.
+
+.. image:: https://res.cloudinary.com/shaka/image/upload/v1542098296/ansible_mailserver/simple_mail_1.png
```
Comment (toadjaune) [inline docs/mail.rst]: I have two remarks on this schema : 
* It's not clear that the first answer from the database is a dashed line
* Haven't you changed "obiwan@jedi" in all of your tests ?

File: docs/mail.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+^^^^^^^^^^^^^^^^^^^
+Within Internet message handling services (MHS), a message transfer agent or
+mail transfer agent (MTA) or mail relay is software that transfers electronic
+mail messages from one computer to another using a client–server application
+architecture. An MTA implements both the client (sending) and server (receiving)
+portions of the Simple Mail Transfer Protocol.
+
+The terms mail server, mail exchanger, and MX host may also refer to a computer
+performing the MTA function. The Domain Name System (DNS) associates a mail
+server to a domain with an MX record containing the domain name of the host(s)
```
Comment (toadjaune) [inline docs/mail.rst]:
```suggestion
server to a domain with a MX record containing the domain name of the host(s)
```

File: docs/mail.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+.. image:: https://res.cloudinary.com/shaka/image/upload/v1542098296/ansible_mailserver/simple_mail_1.png
+
+Components
+----------
+
+Mail Transfer Agent
+^^^^^^^^^^^^^^^^^^^
+Within Internet message handling services (MHS), a message transfer agent or
+mail transfer agent (MTA) or mail relay is software that transfers electronic
```
Comment (toadjaune) [inline docs/mail.rst]: That's a bit heavy.
maybe something like ", a Mail Transfer Agent (MTA, also called Message Tranfer agent, or mail relay) is a software"

File: docs/mail.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+mail messages from one computer to another using a client–server application
+architecture. An MTA implements both the client (sending) and server (receiving)
```
Comment (toadjaune) [inline docs/mail.rst]:
```suggestion
architecture. A MTA implements both the client (sending) and server (receiving)
```

File: docs/mail.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+portions of the Simple Mail Transfer Protocol.
```
Comment (toadjaune) [inline docs/mail.rst]: Rajouter l'acronyme SMTP ?

File: docs/mail.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+The terms mail server, mail exchanger, and MX host may also refer to a computer
+performing the MTA function. The Domain Name System (DNS) associates a mail
+server to a domain with an MX record containing the domain name of the host(s)
+providing MTA services. [Wikipedia_MTA]_
+
+Mail Delivery Agent
+^^^^^^^^^^^^^^^^^^^
+
+A mail delivery agent or message delivery agent (MDA) is a computer software
```
Comment (toadjaune) [inline docs/mail.rst]: Same note than for MTA

File: docs/mail.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+component that is responsible for the delivery of e-mail messages to a local
```
Comment (toadjaune) [inline docs/mail.rst]:
```suggestion
 that is responsible for the delivery of e-mail messages to a local
```

File: docs/mail.rst [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+recipient's mailbox. It is also called a local delivery agent (LDA).
+
+Within the Internet mail architecture, local message delivery is achieved
+through a process of handling messages from the message transfer agent, and
```
Comment (toadjaune) [inline docs/mail.rst]: I think we should start using acronyms (MTA) here.

File: docs/mail.rst [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+performing the MTA function. The Domain Name System (DNS) associates a mail
+server to a domain with an MX record containing the domain name of the host(s)
+providing MTA services. [Wikipedia_MTA]_
+
+Mail Delivery Agent
+^^^^^^^^^^^^^^^^^^^
+
+A mail delivery agent or message delivery agent (MDA) is a computer software
+component that is responsible for the delivery of e-mail messages to a local
+recipient's mailbox. It is also called a local delivery agent (LDA).
```
Comment (toadjaune) [inline docs/mail.rst]: This is probably the best formulation so far for multiple namings

File: docs/mail.rst (hunk 1 above)
Comment (hugoShaka) [reply docs/mail.rst]: I was not clear enough, you reviewed wikipedia :laughing:, it's all my fault.

File: docs/mail.rst (hunk 4 above)
Comment (hugoShaka) [reply docs/mail.rst]: I was not clear enough, you reviewed wikipedia :laughing:, it's all my fault.

File: docs/mail.rst (hunk 1 above)
Comment (hugoShaka) [reply docs/mail.rst]: I corrected both :+1:

File: docs/conf.py [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'Ansible mailserver'
+copyright = '2018, hugoShaka'
```
Comment (HadrienPatte) [inline docs/conf.py]:
```suggestion
copyright = ' %s, hugoShaka' % datetime.date.today().year
```

**Pull request merged**